tsdb/agent/db.go

```diff
@@ -631,17 +631,19 @@ Loop:
 	}
 }
 
-// keepSeriesInWALCheckpoint is used to determine whether a series record should be kept in the checkpoint
+// keepSeriesInWALCheckpointFn returns a function that is used to determine whether a series record should be kept in the checkpoint.
 // last is the last WAL segment that was considered for checkpointing.
```

Comment:
Could perhaps highlight that Agent is using a very different calculation to regular Prometheus, here.